lib/discourse_chat/provider/slack/slack_provider.rb

```diff
@@ -2,13 +2,28 @@
 
 module DiscourseChat::Provider::SlackProvider
   PROVIDER_NAME = "slack".freeze
+  THREAD = "thread".freeze
 
   PROVIDER_ENABLED_SETTING = :chat_integration_slack_enabled
 
   CHANNEL_PARAMETERS = [
                         { key: "identifier", regex: '^[@#]?\S*$', unique: true }
                        ]
 
+  class Engine < ::Rails::Engine
+    isolate_namespace DiscourseChat::Provider::SlackProvider
+  end
+  Topic.register_custom_field_type(DiscourseChat::Provider::SlackProvider::THREAD, :text)
+
+  class ::Topic
+    def set_slack_thread_id(ts)
```

Comment:
We tend not to add `get_` or `set_` at the beginning of ruby function names. I think this would be better as
```suggestion
    def slack_thread_id=(ts)
```
(this will need find/replacing in the other places it is used)

Reply:
Done, thank you! I'm new to Ruby and this is super helpful.